### PR TITLE
Attempt at using only one chunk for solid chunks

### PR DESCRIPTION
**Pull request by petterarvidsson**
@nsg Can you try this out and see if it does anything to help you reduce the memory usage?

### CODE REVIEW

**Comment by nsg** [comment]
@petterarvidsson less then 1G on 19 in view distance (compared to 11G earlier).

File: lib/src/chunk.cpp [hunk 1]
```diff
@@ -12,6 +12,42 @@ namespace konstructs {
...
+            blocks[i].g = (buffer[i * BLOCK_SIZE + 5] & 0xF);
+            blocks[i].b = (buffer[i * BLOCK_SIZE + 5] & 0xF0) >> 4;
+            blocks[i].light = (buffer[i * BLOCK_SIZE + 6] & 0xF);
+            if(blocks[i].type != chunk_type || blocks[i].light > 0 || blocks[i].ambient < AMBIENT_LIGHT_FULL)
+                use_cached = false;
+        }
+        if(use_cached) {
+            try {
+                auto r = cached_data.at(chunk_type);
+                std::cout<<"Got cached!"<<std::endl;
```
Comment (nsg) [inline lib/src/chunk.cpp]: Remember to remove the debug output lines.

**Comment by petterarvidsson** [comment]
@nsg Oki, I am done with the refactoring here now. Wanted to remove shared_ptr from ChunkData since that is now quite wasteful and just makes all the code more complicated. The actual data inside the ChunkData is now what is shared instead.

**Comment by nsg** [comment]
@petterarvidsson looks good :+1:

**Pull request merged**